SudokuSolver/SudokuSolver/Common/BaseSudokuSolver.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.Toolkit.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SudokuSolver.Common
+{
+    public abstract class BaseSudokuSolver : ISudokuSolver
```

Comment:
szczerze mówiąc to bym się zdecydował czy chcesz interfejs czy klasę abstrakcyjną, bo obecnie trochę mija się z celem robienie interfejsu tylko po to żeby implementowała go klasa abstrakcyjna